app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -356,6 +359,7 @@ class HomeFragment : Fragment() {
 
     override fun onDestroyView() {
         super.onDestroyView()
+        _sessionControlInteractor = null
```

Comment:
Interesting, I guess this helps to GC this earlier - I wonder if it's fair to call this a leak in a true sense. Once HomeFragment goes through `onCreateView` again, the old instance of sessionControlInteractor will no longer be referenced, and it will get GC'd. But, that won't happen _until_ user goes back to HomeFragment, so we are retaining this stuff in memory for longer than necessary.

You _could_ similarly null more things here, `currentMode`, `homeFragmentStore`, etc - same logic applies.

Reply:
The main problem with these two objects specifically is that it they retains references to the fragment views that are being destroyed so they are leaking and this can't be GC I believe. Here specifically we are passing in a lambda that references the fragment view and in bookmarks, we pass in a view reference for the snackbar.

Reply:
https://stackoverflow.com/a/59504797